firmware-management/swupdate-handlers/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(swupdate-handlers)
+
+add_library(swupdate-handlers OBJECT rootfs-handler.c arm-handler-common.c)
```

Comment:
> Do we have to put off linking (even to a library) until swupdate links everything together to make sure that swupdate's wierd ctor mechanism works?

Yes... swupdate builds its own libraries in its 'kbuild' system, and we're linking our handler into its 'handlers' shared library/kbuild module. From what I've read, swupdate isn't designed for easily adding your own 'built-in' c handlers and they recommend patching and upstreaming or using their external lua handler system. IIUC, if we wanted to add a shared library, we would have to create a kbuild module with its own set of rules for building, instead of just adding our handler to the swupdate handlers module. This seemed to be the most reasonable way to get it working given how swupdate's build system is structured.

Reply:
Ugh, okay.